### PR TITLE
Small testing related fixes

### PR DESCRIPTION
**Pull request by RedCraig**
This PR replaces #51, which I accidentallied.

* updated README.md to highlight the need for your account keys & that running tests costs money, 
* updated tests to _only_ delete the test database
* tests throw exception if you haven't entered your account key and endpoint. This should make it more obvious when tests fail, previously tests would fail with an encoding error if the key/endpoint had not been entered.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@RedCraig__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

File: test/crud_tests.py [hunk 1]
```diff
@@ -23,8 +23,10 @@
 from struct import *
 from __builtin__ import *
 
+TEST_DB_NAME = 'sample database'
```
Comment (rnagpal) [inline test/crud_tests.py]: Please make TEST_DB_NAME local to class similar to how host and masterKey are defined.

File: test/rate_test.py [hunk 2]
```diff
@@ -8,23 +8,30 @@
...
 import time
-import doctest
+import unittest
 
 import json
 import pydocumentdb.https_connection
 import pydocumentdb.documents as documents
 from collections import deque
 
+import pydocumentdb.https_connection
```
Comment (rnagpal) [inline test/rate_test.py]: import pydocumentdb.https_connection is duplicated.

File: test/rate_test.py [hunk 3]
```diff
@@ -8,23 +8,30 @@
...
+from pydocumentdb import document_client
+
+MASTER_KEY = ''
```
Comment (rnagpal) [inline test/rate_test.py]: There is a masterKey defined inside the RateTest class. You can move this definition of MASTER_KEY inside the class and change the case for HOST as well and use both of them as a class specific variables.

File: test/ttl_tests.py [hunk 4]
```diff
@@ -25,8 +25,9 @@
 import pydocumentdb.document_client as document_client
 import pydocumentdb.errors as errors
 
+TEST_DB_NAME = 'sample database'
```
Comment (rnagpal) [inline test/ttl_tests.py]: Same as other comment, please move this inside the class.

**Comment by rnagpal** [comment]
Made some minor comments. As soon as you incorporate them, I can merge this.

**Comment by RedCraig** [comment]
@rnagpal is there any reason you prefer using class variables in tests? i.e. using `CRUDTests.masterKey` rather than `self.masterKey` when using it in `CRUDTests` test cases?

**Comment by rnagpal** [comment]
Since I started breaking the tests into individual files now, they can be run together in a single command line. In order to avoid accidental overidding of the global variables by different modules since they share the same name, I preferred to have them defined at the class level. This is just a best practice for tests. Let me know if that makes sense.

**Comment by RedCraig** [comment]
AFAIK there's no such thing as a python global which exists across module namespaces. It gets dodgy when you `import *` from a module as it'll import that modules variables (and methods) into the local namespace.

Moving them into a class for tests definitely makes sense, I was just wondering why you referred to them from within that classes methods using the full class name qualifier `CRUDTests.masterKey`, rather than just `self.masterKey`. I don't think there's any difference, so I was curious if you knew one.

**Pull request merged**